repose-aggregator/core/repose-core/src/test/groovy/org/openrepose/nodeservice/httpcomponent/RequestProxyServiceImplTest.groovy

```diff
@@ -161,4 +168,220 @@ class RequestProxyServiceImplTest extends Specification {
         ThreadContext.clearAll()
     }
 
+    def "proxyRequest(host, request, response) will try to use a given connection pool"() {
+        given:
+        when(httpClient.execute(any(HttpUriRequest.class)))
+                .thenReturn(new BasicHttpResponse(HttpVersion.HTTP_1_1, 200, "OK"))
+        when(httpClient.getParams()).thenReturn(new BasicHttpParams())
+
+        when:
+        requestProxyService.proxyRequest("http://www.google.com", new MockHttpServletRequest(), new MockHttpServletResponse())
+
+        and:
+        verify(httpClientService).getClient(null)
+
+        then:
+        noExceptionThrown()
```

Comment:
Urg, this is worse. go back to the old way but just change your method to be:
```def verification(Closure cl) {
     cl()
     true
}```

This will give us mockitos nice error message hopefully.